amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala

```diff
@@ -0,0 +1,35 @@
+package amf.client.environment
+
+import amf.client.remod._
+import amf.plugins.document.vocabularies.metamodel.document.{DialectInstanceModel, DialectModel, VocabularyModel}
+import amf.plugins.document.vocabularies.model.document.{Dialect, DialectInstance, Vocabulary}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+// TODO: ARM remove private[amf]
+private[amf] class AMLClient(protected override val configuration: AMLConfiguration) extends AMFClient(configuration) {
+
+  override implicit val exec: ExecutionContext = configuration.resolvers.executionContext.executionContext
+
+  override def getConfiguration: AMLConfiguration = configuration
+  def parseDialect(url: String): Future[AMLDialectResult] = AMFParser.parse(url, configuration).map {
+    case AMFResult(d: Dialect, r) => new AMLDialectResult(d, r)
+    case other =>
+      throw InvalidBaseUnitTypeException(other.bu.meta.`type`.headOption.map(_.iri()).getOrElse(""),
```

Comment:
Would probably hide the creation of this exception in a private method to avoid polluting that simple method.

Reply:
Solved